src/com/linkedin/parseq/internal/SerialExecutor.java

```diff
@@ -140,12 +157,22 @@ public void run() {
     }
   }
 
+  /**
+   * A priority queue which stores runnables to be executed within a {@link SerialExecutor}.
+   * The implementation has to make sure runnables are sorted in the descending order based
+   * on their priority.
+   */
+  public interface TaskQueue<T extends Prioritizable> {
+    void add(T value);
+    T poll();
+  }
+
   /*
    * Deactivation listener is notified when this executor finished executing a Runnable
    * and there are no other Runnables waiting in queue.
    * It is executed sequentially with respect to other Runnables executed by this Executor.
    */
-  static interface DeactivationListener {
```

Comment:
Because nested interface is static by default and my IDE suggested me to remove the static keyword.